### PR TITLE
Development fix gird tests yml

### PR DESCRIPTION
**Pull request by A-Harby**
### Description

The network variable wasn't assigned any value when the workflow ran by the schedule.

https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1451

### CODE REVIEW

File: .github/workflows/grid_client_tests.yml [hunk 1]
```diff
@@ -2,32 +2,34 @@ name: Grid Client Tests
...
         options: 
-        - 'dev'
-        - 'qa'
-        - 'test'
-        - 'main'
-
+        - "dev"
+        - "qa"
+        - "test"
+        - "main"
```
Comment (zaelgohary) [inline .github/workflows/grid_client_tests.yml:17]: Can we add custom option? We'll also need to provide all urls.

File: .github/workflows/grid_client_tests.yml [hunk 2]
```diff
@@ -2,32 +2,34 @@ name: Grid Client Tests
...
       RMB_PROXY: true
       STORE_SECRET: secret
       MNEMONIC: ${{ secrets.MNEMONIC }}
     steps:
-      - uses: actions/checkout@v2
+      - name: set network for Schedule
+        if: ${{env.NETWORK}} == ""
+        run: echo "NETWORK=dev" >> $GITHUB_ENV
+      - uses: actions/checkout@v4
       - name: Set up node 18
```
Comment (zaelgohary) [inline .github/workflows/grid_client_tests.yml:31]: Maybe change this to Set up node version?

**Review by zaelgohary — CHANGES_REQUESTED** [review]

File: .github/workflows/grid_client_tests.yml (hunk 2 above)
Comment (A-Harby) [reply .github/workflows/grid_client_tests.yml]: Changed in [195c3d3](https://github.com/threefoldtech/tfgrid-sdk-ts/pull/1454/commits/195c3d369550071f470b1d4e66a524b74af13cc0)

File: .github/workflows/grid_client_tests.yml (hunk 1 above)
Comment (A-Harby) [reply .github/workflows/grid_client_tests.yml:17]: The workflow is only for the default setups though, and any custom options would need to be manually run. These tests are meant to run in basic environments.

**Comment by zaelgohary** [comment]
@A-Harby  I edited the workflow to test it and ran it twice and it failed both times. You can check the logs [here](https://github.com/threefoldtech/tfgrid-sdk-ts/actions/runs/6971016372/job/18971030282).

**Comment by A-Harby** [comment]
> @A-Harby I edited the workflow to test it and ran it twice and it failed both times. You can check the logs [here](https://github.com/threefoldtech/tfgrid-sdk-ts/actions/runs/6971016372/job/18971030282).

The [issue](https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1451) I tried to solve was that the tests themselves didn't run, giving the error: `Couldn't create TFClient without store secret` and the cause of it was that the network wasn't assigned a value.

And from your logs, it shows that the issue is fixed because the tests started to run.

But the fail test from logs was caused by this error below, and it is fixed on https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1412, but it was merged to branch 3.13, and workflow runs on the latest development branch, which is why it's still failing.
image

**Review by zaelgohary — APPROVED** [review]

**Pull request merged**